core/templates/pages/exploration-editor-page/settings-tab/settings-tab.component.spec.ts

```diff
@@ -255,33 +255,42 @@ describe('Settings Tab Component', () => {
     });
 
     it('should initialize controller properties after its initialization',
-      () => {
+      (done) => {
         expect(ctrl.isRolesFormOpen).toBe(false);
         expect(ctrl.canDelete).toBe(true);
         expect(ctrl.canModifyRoles).toBe(true);
         expect(ctrl.canReleaseOwnership).toBe(true);
         expect(ctrl.canUnpublish).toBe(true);
         expect(ctrl.explorationId).toBe(explorationId);
         expect(ctrl.canManageVoiceArtist).toBe(true);
+        expect(ctrl.loggedInUser).toBe('username1');
 
-        expect(ctrl.CATEGORY_LIST_FOR_SELECT2[0]).toEqual({
-          id: 'Astrology',
-          text: 'Astrology'
-        });
+        setTimeout(()=>{
+          $scope.$apply();
 
-        expect(ctrl.stateNames).toEqual(['Introduction']);
-        expect(ctrl.hasPageLoaded).toBe(true);
-        expect(ctrl.loggedInUser).toBe('username1');
+          expect(ctrl.CATEGORY_LIST_FOR_SELECT2[0]).toEqual({
+            id: 'Astrology',
+            text: 'Astrology'
+          });
+
+          expect(ctrl.stateNames).toEqual(['Introduction']);
+          expect(ctrl.hasPageLoaded).toBe(true);
+
+          done();
+        }, 501);
       });
 
     it('should refresh settings tab when refreshSettingsTab flag is ' +
-        'broadcasted', () => {
+        'broadcasted', fakeAsync(() => {
       routerService.onRefreshSettingsTab.emit();
       $scope.$apply();
 
+      tick(1000);
```

Comment:
why is this 1000 if the setTimeout is only 500?

Reply:
This was added when I was a timeout of 1000 was used. Looks like I forgot to edit it. I have fixed it now.